fix(compiler-cli): capture metadata for undecorated fields

crisbeto opened the pull request:
Currently if `TestBed.overrideComponent` is used on a class that uses initializer APIs (e.g. `input()`), the initializer metadata will be wiped out, because `overrideComponent` re-compiles the class with the information set by `setClassMetadata`. `setClassMetadata` only captures decorated members at the moment.

These changes introduce some logic to capture the new initializer-based APIs in `setClassMetadata` as well.

Fixes #57944.

crisbeto committed:
refactor(compiler-cli): track member metadata using output AST

Reworks the logic that tracks the decorator metadata for members to do so using the output AST, rather than wrapping the TypeScript AST. This makes it easier to programmatically generate new members that weren't part of the TypeScript AST before.

crisbeto committed:
refactor(compiler): expose member decorator identifiers

Adds the member decorators to the map of exposed compiler symbols.

devversion reviewed (approved):
LGTM, non-blocking comments

crisbeto committed:
fix(compiler-cli): capture metadata for undecorated fields

Currently if `TestBed.overrideComponent` is used on a class that uses initializer APIs (e.g. `input()`), the initializer metadata will be wiped out, because `overrideComponent` re-compiles the class with the information set by `setClassMetadata`. `setClassMetadata` only captures decorated members at the moment.

These changes introduce some logic to capture the new initializer-based APIs in `setClassMetadata` as well.

Fixes #57944.

thePunderWoman commented:
@crisbeto We had to revert this due to a test breakage in G3.

crisbeto commented:
Will open a new PR.
